src/Main.elm

```diff
@@ -628,21 +653,108 @@ raisedDiv =
         ]
 
 
-modalView : Model -> Html Msg
-modalView model =
+type Menu
+    = DifficultyMenu Difficulty GridProperties
```

Comment:
the idea for this `Menu`, based on the Windows screenshot i had posted, is you have a selected `Difficulty` and you have three fields held in `GridProperties`. those fields don't necessarily reflect the selected `Difficulty`. they are just form values used for construction a `Difficulty` with `Custom` once the time comes.

as you'll see down below, i'm just disabling the inputs until you select `Custom`.